pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,490 @@
+        function = cur_stat.b.function
+        inv_modes = {v: k for k, v in cls.MODES.items()}
+        cur_mode = inv_modes["F"+str(function)]
+        rnge = cur_stat.b.range
+        cur_range = list(cls.RANGES[cur_mode].keys())[rnge-1]
+        return cur_range
+
+    @staticmethod
+    def decode_trigger(status_bytes):
+        """Method to decode trigger mode
+
+        :param status_bytes:   list of bytes to be decoded
+        :return trigger_mode: string with the current trigger mode
+        :rtype trigger_mode: str
+
+        """
+        cur_stat= Status(Status_bytes(*status_bytes.encode(encoding="ASCII")))
+        i_trig = cur_stat.b.int_trig
+        e_trig = cur_stat.b.ext_trig
+        if i_trig == 0:
+            if e_trig == 0:
+                trigger_mode = "hold"
+            else:
+                trigger_mode = "external"
+        else:
+            trigger_mode = "internal"
+        return trigger_mode
+
+    #commands/properties for instrument control
+    active_connectors = Instrument.measurement(
+        "B",
+        """Return selected connectors ("front"/"back"), based on front-panel selector switch
+        """,
+        get_process = (lambda x: HP3478A.decode_status(x, "front_rear")),
+        values={"back":0, "front":1},
+        map_values = True,
+        )
+
+    auto_range_enabled = Instrument.measurement(
+        "B",
+        """ Return auto-ranging status, returns False if manual range and True if auto-range active.
+        For manual range control the range property can be set
+        """,
+        get_process = (lambda x: HP3478A.decode_status(x, "auto_range")),
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    auto_zero_enabled = Instrument.control(
+        "B",
+        "Z%d",
+        """ Returns autozero settings on the HP3478, this property can be set
+        (False: disabled, True: enabled)
+        """,
+        get_process = (lambda x: HP3478A.decode_status(x, "auto_zero")),
+        validator = strict_discrete_set,
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    calibration_enabled = Instrument.measurement(
+        "B",
+        """Return calibration enable switch setting (False: cal disabled, True: cablibration possible),
+        based on front-panel selector switch
+        """,
+        get_process = (lambda x: HP3478A.decode_status(x, "cal_enable")),
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    def check_errors(self):
+        """
+        Method to read the error status register
+
+        :return error_status: one byte with the error status register content
+        :rtype error_status: int
+        """
+
+        if self.error_status != 0:
+            log.error("HP3478A error detected: $s", self.ERRORS[self.error_status])
+        return self.error_status
+
+    error_status = Instrument.measurement(
```

Comment:
Just FYI: This would normally be an Instrument method `check_errors()` that could also be used by other properties to check errors after operation.